packages/commerce-sdk-react/src/auth/index.ts

```diff
@@ -0,0 +1,297 @@
+import {helpers, ShopperLogin, ShopperLoginTypes} from 'commerce-sdk-isomorphic'
+import jwtDecode from 'jwt-decode'
+import {ApiClientConfigParams} from '../hooks/types'
+import {BaseStorage, LocalStorage, CookieStorage} from './storage'
+
+type Helpers = typeof helpers
+interface AuthConfig {
+    shortCode: string
+    organizationId: string
+    clientId: string
+    siteId: string
+    redirectURI: string
+    proxy: string
+}
+
+interface JWTHeaders {
+    exp: number
+    iat: number
+}
+
+// this type is slightly different from ShopperLoginTypes.TokenResponse, reasons:
+// 1. TokenResponse is too generic (with & {[key:string]: any}), we need a more
+//    restrictive type to make sure type safe
+// 2. The refresh tokens are stored separately for guest and registered user. Instead
+//    of refresh_token, we have refresh_token_guest and refresh_token_registered
+type AuthDataKeys =
+    | 'access_token'
+    | 'customer_id'
+    | 'enc_user_id'
+    | 'expires_in'
+    | 'id_token'
+    | 'idp_access_token'
+    | 'refresh_token_guest'
+    | 'refresh_token_registered'
+    | 'token_type'
+    | 'usid'
+type AuthDataMap = Record<
+    AuthDataKeys,
+    {
+        storage: BaseStorage
+        key: string
+        callback?: () => void
+    }
+>
+
+/**
+ *
+ * @Internal
+ */
+class Auth {
+    private client: ShopperLogin<ApiClientConfigParams>
+    private redirectURI: string
+    private _onClient = typeof window !== 'undefined'
+    private pending: Promise<ShopperLoginTypes.TokenResponse> | undefined
+    private localStorage: LocalStorage
+    private cookieStorage: CookieStorage
+    private REFRESH_TOKEN_EXPIRATION_DAYS = 90
```

Comment:
Q: Is the tokens expiration encoded within it. Do we need to explicitly create this constant?

Reply:
The token expiry is not encoded within the refresh token. (it's not a JWT).